server.py

```diff
@@ -228,3 +231,42 @@ def semesterIDType(reqstr):
 
   # Convert Officer objects back to dictionaries before returning
   return jsonify([x.__dict__ for x in officers])
+
+#I'm keeping these global variables here for now
+# so it's easier to determine that it goes with the events api
+# when the pieces of the api are inevitably split up
+EV = []
+EVENTS_CACHED = datetime.now()
+def cacheEvents():
+  """
+  Caches the event calendar.
+  """
+  calurl = 'https://calendar.google.com/calendar/ical/ca149os3pmnh0dcopr1jn2negg%40group.calendar.google.com/public/basic.ics'
+  dtstrformat = '%Y-%m-%dT%H:%M:%S.%fZ'
+  global EV, EVENTS_CACHED
+
+  try:
+    cal_request = requests.get(calurl)
+    cal_request.raise_for_status() #requests doesn't raise HTTPError on its own
+    gcal = Calendar.from_ical(cal_request.content)
+    EV = []
+    for comp in [x for x in gcal.walk() if x.name == "VEVENT"]:
+      EV.append({'summary': comp.decoded('SUMMARY').decode('UTF-8'),
+                 'timeStart': comp.decoded('DTSTART').strftime(dtstrformat),
+                 'timeEnd': comp.decoded('DTEND').strftime(dtstrformat),
+                 'location': comp.decoded('LOCATION').decode('UTF-8'),
+                 'description': comp.decoded('DESCRIPTION').decode('UTF-8')})
+    EVENTS_CACHED = datetime.now()
+  except requests.HTTPError:
```

Comment:
I feel like this try/except encloses too much. It's just requests.raise_for_status that throws this, right? I think you'll also need to declare cal_request outside the block in order to use it outside. Also note to self that we need a real logging system for this error.

Reply:
Actually, if we're handling it here, do we even need a try block? Can we just return early if status isn't 200?

Reply:
I take that back, we can't handle this here. We need to return an error code to the frontend if this function fails.

Reply:
Yeah, I was going to implement that but I'm not sure what format to send the error code in. Does Flask have a specific implementation for this?

Reason the try block encloses so much is because everything after the "raise_for_status" call depends on the request succeeding.

Reply:
Right, but then raise_for_status could be in a try-catch by itself - catch the http exception, see if there's old cache data we can serve, and if not throw another exception which gets caught in `getEvents()` itself

Reply:
For how to throw errors, I do this in `getOfficers`, looks like:
```python
return jsonify({"error": "Invalid parameter 'semester'"}), 400
```
which is a tuple of `(json payload, status code)`. The json can be whatever, but it should probably at least have an "error" field which describes what went wrong. The status code should be a number from [this list](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes), probably 500 or 502 in this case. To be clear, this would be returned from `getEvents`, not here - like try-catch `cacheEvents` inside `getEvents` and return this from there.